src/types/num.rs

```diff
@@ -420,6 +422,49 @@ mod bigint_conversion {
             let zero: BigInt = FromPyObject::extract(fib.call1("fib", (0,)).unwrap()).unwrap();
             assert_eq!(zero, BigInt::from(0));
         }
+
+        /// `OverflowError` on converting python int to BigInt
+        #[test]
+        fn issue_629() {
+            let gil = Python::acquire_gil();
+            let py = gil.python();
+            fn test<T>(value: T, py: Python)
+            where
+                T: 'static
+                    + Clone
+                    + ToPyObject
+                    + for<'a> FromPyObject<'a>
+                    + Eq
+                    + fmt::Display
+                    + fmt::Debug,
+            {
+                let type_id = TypeId::of::<T>();
+                let type_name = if type_id == TypeId::of::<BigInt>() {
+                    "BigInt"
+                } else {
+                    assert_eq!(type_id, TypeId::of::<BigUint>());
+                    "BigUint"
+                };
+                println!("{}: {}", type_name, value);
+                let python_value = value.clone().to_object(py);
+                let roundtrip_value = python_value.extract::<T>(py).unwrap();
+                assert_eq!(value, roundtrip_value);
+            }
+            for i in 0..=256usize {
+                test(BigInt::from(i), py);
```

Comment:
Overflow occurs when `x` in `BifInt::from(x)` consists of 8 * n bits, right?
Then these tests don't look very effective.

But I also did not come up with good test cases, so it's OK if you don't mind.

Reply:
The idea is that the tests weren't effective before, so we should test lots of stuff to ensure we didn't miss anything.

Reply:
everything in `128 <= v && v <= 255` triggers the bug, as well as every 8th value for the tests with `<<`

Reply:
added comment to that effect